File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.cpp
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {

        unordered_set<int> set1(nums1.begin(), nums1.end());
        unordered_set<int> set2(nums2.begin(), nums2.end());

        vector<int> difference1;
        vector<int> difference2;

        // Elements in nums1 but not in nums2
        for (int num : set1) {
            if (set2.find(num) == set2.end()) {
                difference1.push_back(num);
            }
        }

        // Elements in nums2 but not in nums1
        for (int num : set2) {
            if (set1.find(num) == set1.end()) {
                difference2.push_back(num);
            }
        }

        return {difference1, difference2};
    }
};

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.java
import java.util.*;

class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        // Create sets to store unique elements of each array
        Set<Integer> set1 = new HashSet<>();
        for (int num : nums1) set1.add(num);
        
        Set<Integer> set2 = new HashSet<>();
        for (int num : nums2) set2.add(num);

        List<Integer> difference1 = new ArrayList<>();
        List<Integer> difference2 = new ArrayList<>();

        // Find elements in nums1 not in set2
        for (int num : set1) {
            if (!set2.contains(num)) {
                difference1.add(num);
            }
        }

        // Find elements in nums2 not in set1
        for (int num : set2) {
            if (!set1.contains(num)) {
                difference2.add(num);
            }
        }

        // Return as a list of lists
        return Arrays.asList(difference1, difference2);
    }
}